test/test_grids.jl

```diff
@@ -290,8 +290,11 @@ end
 
 function test_grid_equality_over_architectures()
     grid_cpu = RectilinearGrid(CPU(), topology=(Periodic, Periodic, Bounded), size=(3, 7, 9), x=(0, 1), y=(-1, 1), z=0:9)
-    grid_gpu = RectilinearGrid(GPU(), topology=(Periodic, Periodic, Bounded), size=(3, 7, 9), x=(0, 1), y=(-1, 1), z=0:9)
-
+    if CUDA.has_cuda()
+        grid_gpu = RectilinearGrid(CUDAGPU(), topology=(Periodic, Periodic, Bounded), size=(3, 7, 9), x=(0, 1), y=(-1, 1), z=0:9)
+    elseif AMDGPU.has_rocm_gpu()
+        grid_gpu = RectilinearGrid(ROCMGPU(), topology=(Periodic, Periodic, Bounded), size=(3, 7, 9), x=(0, 1), y=(-1, 1), z=0:9)
+    end
     return grid_cpu == grid_gpu
 end
```

Comment:
Here, it may make more sense to pass in the "target" architecture:

```julia
function test_grid_equality_over_architectures(arch)
     grid_cpu = RectilinearGrid(CPU(), topology=(Periodic, Periodic, Bounded), size=(3, 7, 9), x=(0, 1), y=(-1, 1), z=0:9)
     grid_gpu = RectilinearGrid(GPU(), topology=(Periodic, Periodic, Bounded), size=(3, 7, 9), x=(0, 1), y=(-1, 1), z=0:9)

      grid_arch = RectilinearGrid(arch, topology=(Periodic, Periodic, Bounded), size=(3, 7, 9), x=(0, 1), y=(-1, 1), z=0:9)

     return grid_cpu == grid_arch
 end
```

Reply:
then we can control the "target" `arch` outside this function.

Reply:
This was added in the new commits.